chunks/lang-node/Dockerfile

```diff
@@ -9,12 +9,16 @@ USER gitpod
 ENV TRIGGER_REBUILD=1
 ENV NODE_VERSION=${NODE_VERSION}
 
+ENV PNPM_HOME=/home/gitpod/.pnpm
+ENV PATH=/home/gitpod/.nvm/versions/node/v${NODE_VERSION}/bin:/home/gitpod/.yarn/bin:${PNPM_HOME}:$PATH
+
 RUN curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | PROFILE=/dev/null bash \
     && bash -c ". .nvm/nvm.sh \
         && nvm install v${NODE_VERSION} \
         && nvm alias default v${NODE_VERSION} \
-        && npm install -g typescript yarn pnpm node-gyp" \
+        && npm install -g yarn \
+        && yarn global add typescript yarn pnpm node-gyp" \
```

Comment:
Please notice, this is intentional, for global packages survive across nvm versions.

Reply:
I'm not exactly familiar with js tooling. But I have a question, does it mean that if I switch to a different nodejs version with `nvm`, `typescript yarn pnpm node-gyp` executables will be using the old version interpreter to operate unless the user manually reinstall them in the changed version?

Reply:
Good point, @axonasif. 

@gengjiawen is this a standard practice, to install yarn globally, and then use yarn to install other packages globally? If yes, could we ask you to share a reference or two to help us better understand the trade-offs?

I don't see any references to `yarn global ...` [here](https://github.com/gitpod-io/gitpod), so it's not necessarily a standard that we're using "now".

Perhaps @gitpod-io/engineering-webapp can help here, as they work a bunch in node/typescript/yarn, too.

Reply:
> But I have a question, does it mean that if I switch to a different nodejs version with nvm, typescript yarn pnpm node-gyp executables will be using the old version interpreter to operate unless the user manually reinstall them in the changed version?

Nope, those packages works as expected. They will be still work if you uninstall preinstalled nodejs.

> is this a standard practice, to install yarn globally

Yes. Same as pnpm and npm.

> then use yarn to install other packages globally? 

yeap. npm can do the same job. But nvm hijacked npm, so it only partly works.

> I don't see any references to yarn global ... [here](https://github.com/gitpod-io/gitpod), so it's not necessarily a standard that we're using "now".

Yeap. It's not needed if don't switch to another Node.js version or use yarn or pnpm install global packages.
But if you do, it's broken.

Reply:
Generally speaking, it's more robust and compatible with less surprise with current patch.

Reply:
@axonasif ping

Reply:
:wave: @gengjiawen I've had time to think about this change and scenario.

Please refer to [this matrix](https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3) for an example. You can see it relates compatible versions of Angular, Node, Typescript, and RxJS. So, users may wish to use nvm to switch between node/npm installations that are coupled to distinct Typescript installations (rather than have Typescript be installed globally across all node/npm versions). 

Given the above example, it would be more desirable to install Typescript with NPM, instead of yarn, so that they can use NVM to swap between node versions (and their related global dependencies, like Typescript).

If it's desired to use yarn to manage global dependencies (across all node versions which you can switch between node version using NVM), I recommend creating a custom base image, and manage that related flow from your own [custom Dockerfile](https://www.gitpod.io/docs/references/gitpod-dockerfile#custom-docker-image).

Reply:
Given :point_up:, unless we can reach an alternate solution, this PR will have to be closed w/o merge.

Reply:
@AlexTugarev you introduced nvm originally IIRC. Could you review? :thinking: